AdvancedTeleport-Bukkit/src/main/java/io/github/niestrat99/advancedteleport/api/ATPlayer.java

```diff
@@ -180,7 +179,7 @@ public void setTeleportationEnabled(boolean teleportationEnabled, SQLManager.SQL
      * @param teleportationEnabled true to enable teleportation, false to disable it.
      * @return a completable future of whether the action failed or succeeded.
      */
-    public CompletableFuture<Void> setTeleportationEnabled(boolean teleportationEnabled) {
+    public CompletableFuture<Boolean> setTeleportationEnabled(boolean teleportationEnabled) {
```

Comment:
I changed this from Boolean to Void to have exceptions properly handled within the handleAsync function, the Boolean just indicates whether the operation was successful (true if no errors, false if no errors). Would you recommend it any particular way?

Reply:
Same goes for any other method signatures changed to reflect this.

Reply:
I've made a kinda simplified way of handling the errors which i think is cleaner and can also allow for proving more detailed error details via the created exceptions.
Honestly i don't have much experience with futures and i mainly work in kotlin with coroutines and the arrow functional framework for representing values that can fail with the either interface.

Reply:
At a glance, it looks a lot tidier! Thank you for covering that. 

It's absolutely fine, I believe I started using the method in one of my standalone programs which otherwise resulted in missing exceptions that would lead to botched logic conclusions. It may be slightly different with AT due to the use of the async executor but the more you know.